src/dataflow-types/src/client/controller.rs

```diff
@@ -138,5 +146,48 @@ pub enum ReadPolicy<T> {
     /// This function will only be re-evaluated when the write frontier changes.
     /// If the intended behavior is to change in response to external signals,
     /// consider using the `ValidFrom` variant to manually pilot compaction.
-    LagWriteFrontier(Box<dyn Fn(AntichainRef<T>) -> Antichain<T>>),
+    ///
+    /// The `Arc` makes the function cloneable.
+    LagWriteFrontier(Arc<dyn Fn(AntichainRef<T>) -> Antichain<T>>),
+    /// Allows one to express multiple read policies, taking the least of
+    /// the resulting frontiers.
```

Comment:
@antiguru observes that this will eventually be annoying to serialize, and I think it could just be a `lag: mz_repr::Timestamp`. Doing this made a stack of other things "not generic in `T`" which .. I'll try and figure out in the future.

Reply:
Let's keep it as-is and if we really want to serialize the `ReadPolicy`, we'll find a way to fix it.